Update the W/X quickstart articles when the Dev Kit GA

MingjiaLiu1995 opened the pull request:
1. Add the Dev Kit option in the Excel quickstart article.
2. Add the Dev Kit option in the Word quickstart article.

The articles when NOT be published until the Dev Kit is ready for GA.

MingjiaLiu1995 commented on docs/images/office-add-ins-development-kit-samplepage-excelquickstart.png:
The snapshoots are now placeholders. Need to update with the new sample naming

MingjiaLiu1995 replied:
Updated

MingjiaLiu1995 commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+You can create an Office Add-in by using the Office Add-ins Development Kit or the Yeoman generator in Visual Studio Code. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally.
+
+# [Office Add-ins Development Kit](#tab/devkit)
+## Prerequisites
```
The prerequisite section is same as the one section in (docs/develop/development-kit-overview.md)

MingjiaLiu1995 commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+> [!NOTE]
+> You will receive prompts to install the Office Add-ins Development Kit extension if you haven't. You can also [install the Office Add-ins Development Kit manually](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/development-kit-overview?tabs=vscode).
+
+## Explore the project
```
The "Explore the project"section is most similar as the one in the yeoman option.

AlexJerabek replied:
Are they the same? If so, we should make an [include file](https://review.learn.microsoft.com/en-us/help/platform/includes-best-practices?branch=main).

MingjiaLiu1995 commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+1. The ./src/taskpane/taskpane.css file contains the CSS that's applied to content in the task pane.
+1. The ./src/taskpane/taskpane.js file contains the Office JavaScript API code that facilitates interaction between the task pane and the Office client application.
+
+## Try it out
```
"Try it out", "Stop testing", "Troubleshooting" are same as the ones in (docs/develop/development-kit-overview.md)

AlexJerabek replied:
Same suggestion for an include file. This prevents content from being different in two places if we miss something in an update.

AlexJerabek replied:
Is there a reason we didn't include the headers in the include files? I think you could make this whole thing one include file.

MingjiaLiu1995 replied:
For the "start debugging" one, there is other content after the header in the dev kit overview doc, so the header is not in the include file. Others are included.

MingjiaLiu1995 commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+
+Click the following button to create an add-in project using the Office Add-ins Development Kit for Visual Studio Code. A page that contains the project decription will open in Visual Studio Code.
+
+<button type="button">[Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Excel-HelloWorld-TaskPane-JS)</button>
```
I imagine a clickable button to link to the sample page in VSC. Can we achieve that in MS Learn site?

AlexJerabek replied:
I think this is the best way to do that: https://review.learn.microsoft.com/en-us/help/platform/markdown-reference?branch=main#next-step-action. I've added the markdown as a suggestion.

MingjiaLiu1995 reviewed (commented):
Hi @AlexJerabek , I start a initial proposal on the documentation update to prepare for the Dev Kit GA. Please feel free to let me know your thoughts! Thanks ;)

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
 # Build an Excel task pane add-in
 
-In this article, you'll walk through the process of building an Excel task pane add-in with Yo Office.
+In this article, you'll walk through the process of building an Excel task pane add-in.
 
+You can create an Office Add-in by using the Office Add-ins Development Kit or the Yeoman generator in Visual Studio Code. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally.
```
```suggestion
In this article, you'll walk through the process of building an Excel task pane add-in. You'll use either the Office Add-ins Development Kit or the Yeoman generator to create your Office Add-in. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally. If you'd like to use Visual Studio Code, we recommend the Office Add-ins Development Kit.
```

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+
+Click the following button to create an add-in project using the Office Add-ins Development Kit for Visual Studio Code. A page that contains the project decription will open in Visual Studio Code.
+
+<button type="button">[Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Excel-HelloWorld-TaskPane-JS)</button>
```
```suggestion
> [!div class="nextstepaction"]
> [Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Excel-HelloWorld-TaskPane-JS)
```

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+
+<button type="button">[Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Excel-HelloWorld-TaskPane-JS)</button>
+
+<img src="../images/office-add-ins-development-kit-samplepage-excelquickstart.png" alt="Screenshot shows the sample page in Visual Studio Code"/>
+
+In the prompted page, select **Create** to create the add-in project. In the **Workspace folder** dialog that opens, select the folder where you want to create the project. 
```
I'd move the screenshot so it follow the "Create" instruction. In the screenshot, surround the Create button with a red box to highlight where it is in the UI.

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+
+In the prompted page, select **Create** to create the add-in project. In the **Workspace folder** dialog that opens, select the folder where you want to create the project. 
+
+Office Add-ins Development Kit will create the project. It will then open the project *in a second Visual Studio Code window*. Close the original Visual Studio Code window.
```
I think limiting the italics to the keyword will make it more obvious for readers.
```suggestion
The Office Add-ins Development Kit will create the project. It will then open the project in a *second* Visual Studio Code window. Close the original Visual Studio Code window.
```

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,71 @@ ms.localizationpriority: high
+Office Add-ins Development Kit will create the project. It will then open the project *in a second Visual Studio Code window*. Close the original Visual Studio Code window.
+
+> [!NOTE]
+> You will receive prompts to install the Office Add-ins Development Kit extension if you haven't. You can also [install the Office Add-ins Development Kit manually](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/development-kit-overview?tabs=vscode).
```
```suggestion
> You'll be prompted to install the Office Add-ins Development Kit extension if don't already have it. If you need to install the Office Add-ins Development Kit manually, follow [these steps](../development-kit-overview.md?tabs=vscode).
```

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -10,6 +10,69 @@ ms.localizationpriority: high
 
 In this article, you'll walk through the process of building a Word task pane add-in.
 
+You can create an Office Add-in by using the Office Add-ins Development Kit or the Yeoman generator in Visual Studio Code. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally.
```
```suggestion
In this article, you'll walk through the process of building a Word task pane add-in. You'll use either the Office Add-ins Development Kit or the Yeoman generator to create your Office Add-in. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally. If you'd like to use Visual Studio Code, we recommend the Office Add-ins Development Kit.
```

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -10,6 +10,69 @@ ms.localizationpriority: high
+
+Click the following button to create an add-in project using the Office Add-ins Development Kit for Visual Studio Code. A page that contains the project decription will open in Visual Studio Code.
+
+<button type="button">[Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Word-HelloWorld-TaskPane-JS)</button>
```
```suggestion
> [!div class="nextstepaction"]
> [Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Excel-HelloWorld-TaskPane-JS)
```

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -10,6 +10,69 @@ ms.localizationpriority: high
+
+<button type="button">[Create an add-in in Visual Studio Code](vscode://msoffice.microsoft-office-add-in-debugger/open-specific-sample?sample-id=Word-HelloWorld-TaskPane-JS)</button>
+
+<img src="../images/office-add-ins-development-kit-samplepage-wordquickstart.png" alt="Screenshot shows the sample page in Visual Studio Code"/>
```
Same suggestion as with Excel for a different screenshot.

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -10,6 +10,69 @@ ms.localizationpriority: high
+
+In the prompted page, select **Create** to create the add-in project. In the **Workspace folder** dialog that opens, select the folder where you want to create the project. 
+
+Office Add-ins Development Kit will create the project. It will then open the project *in a second Visual Studio Code window*. Close the original Visual Studio Code window.
```
```suggestion
The Office Add-ins Development Kit will create the project. It will then open the project in a *second* Visual Studio Code window. Close the original Visual Studio Code window.
```

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -10,6 +10,69 @@ ms.localizationpriority: high
+Office Add-ins Development Kit will create the project. It will then open the project *in a second Visual Studio Code window*. Close the original Visual Studio Code window.
+
+> [!NOTE]
+> You will receive prompts to install the Office Add-ins Development Kit extension if you haven't. You can also [install the Office Add-ins Development Kit manually](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/development-kit-overview?tabs=vscode).
```
```suggestion
> You'll be prompted to install the Office Add-ins Development Kit extension if don't already have it. If you need to install the Office Add-ins Development Kit manually, follow [these steps](../development-kit-overview.md?tabs=vscode).
```

AlexJerabek commented:
Thanks for putting this together @MingjiaLiu1995! When are we planning to launch the Dev Kit for GA?

MingjiaLiu1995 commented:
> Thanks for putting this together @MingjiaLiu1995! When are we planning to launch the Dev Kit for GA?

Thanks for the review comments Alex! 

The only GA dependency is completing merging all the existing samples. Herman submitted the last sample repo today(Thursday), so I guess we can target some day in the next 1~2 weeks if there is no other concern? 

Btw, who would be the dev kit contact during your oof? I'll address all these comments tomorrow(Friday), so hopefully only small adjustment and review are needed after. (And wish you have a great vacation!)

MingjiaLiu1995 commented:
@AlexJerabek I have addressed all the previous comments. Let me know if there are more. Thanks!

AlexJerabek commented on docs/includes/devkit-project-components-taskpane.md:
```diff
@@ -0,0 +1,6 @@
+The add-in project that you've created with the Office Add-ins Development Kit contains sample code for a basic task pane add-in. If you'd like to explore the components of your add-in project, open the project in your code editor and review the files listed below. When you're ready to try out your add-in, proceed to the next section.
+
+1. The ./manifest.xml or manifest.json file in the root directory of the project defines the settings and capabilities of the add-in.
+1. The ./src/taskpane/taskpane.html file contains the HTML markup for the task pane.
+1. The ./src/taskpane/taskpane.css file contains the CSS that's applied to content in the task pane.
+1. The ./src/taskpane/taskpane.js file contains the Office JavaScript API code that facilitates interaction between the task pane and the Office client application.
```
File names should be in bold
```suggestion
1. The **./manifest.xml** or **./manifest.json** file in the root directory of the project defines the settings and capabilities of the add-in.
1. The **./src/taskpane/taskpane.html** file contains the HTML markup for the task pane.
1. The **./src/taskpane/taskpane.css** file contains the CSS that's applied to content in the task pane.
1. The **./src/taskpane/taskpane.js** file contains the Office JavaScript API code that facilitates interaction between the task pane and the Office client application.
```

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,47 @@ ms.localizationpriority: high
+
+In the prompted page, select **Create** to create the add-in project. In the **Workspace folder** dialog that opens, select the folder where you want to create the project. 
+
+<img src="../images/office-add-ins-development-kit-samplepage-excelquickstart.png" alt="Screenshot shows the sample page in Visual Studio Code"/>
```
Favor the [Learn image markdown syntax](https://review.learn.microsoft.com/en-us/help/platform/markdown-reference?branch=main#custom-learn-markdown-for-standard-conceptual-images-recommended)
```suggestion
:::image type="content" source="../images/office-add-ins-development-kit-samplepage-excelquickstart.png" alt-text="The Development Kit sample gallery in Visual Studio Code":::
```

AlexJerabek commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,47 @@ ms.localizationpriority: high
+## Troubleshooting
+
+[!include[Dev_kit_troubleshooting](../includes/dev-kit-troubleshooting.md)]
+
+
```
```suggestion

```

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -8,8 +8,47 @@ ms.localizationpriority: high
+## Troubleshooting
+
+[!include[Dev_kit_troubleshooting](../includes/dev-kit-troubleshooting.md)]
+
+
```
```suggestion

```

AlexJerabek commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -8,8 +8,47 @@ ms.localizationpriority: high
+
+In the prompted page, select **Create** to create the add-in project. In the **Workspace folder** dialog that opens, select the folder where you want to create the project. 
+
+<img src="../images/office-add-ins-development-kit-samplepage-wordquickstart.png" alt="Screenshot shows the sample page in Visual Studio Code"/>
```
```suggestion
:::image type="content" source="../images/office-add-ins-development-kit-samplepage-wordquickstart.png" alt-text="The Development Kit sample gallery in Visual Studio Code":::
```

Rick-Kirkham commented:
> I believe I have addressed all comments.. (it's weird that it says changes requested, but I can't find the suggestion)

The "Changes requested" doesn't go away until Alex re-reviews. When do these changes need to publish?

Rick-Kirkham commented on docs/includes/dev-kit-start-debugging.md:
```diff
@@ -0,0 +1,5 @@
+1. Open the extension by selecting the Office Add-ins Development Kit icon in the **Activity Bar**.
+1. Select **Preview Your Office Add-in (F5)**
+1. In the Quick Pick menu, select the option **{Office Host} Desktop (Edge Chromium)**. This will launch the add-in and debug the code.
```
```suggestion
1. In the Quick Pick menu, select the option **{Office Application} Desktop (Edge Chromium)**, where '{Office Application}' is the appropriate application, such as "Excel" or "Word". This will launch the add-in and debug the code.
```

Rick-Kirkham commented on docs/includes/dev-kit-start-debugging.md:
```diff
@@ -0,0 +1,5 @@
+1. Select **Preview Your Office Add-in (F5)**
+1. In the Quick Pick menu, select the option **{Office Host} Desktop (Edge Chromium)**. This will launch the add-in and debug the code.
+
+The development kit checks that the prerequisites are met before debugging starts. Check the terminal for detailed information if there are issues with your environment. After this process, the Office desktop application launches and sideloads the add-in. Please note that the first time you run a project, it may make take a few minutes to install the dependencies. You will need to install the cer ...
```
```suggestion
The development kit checks that the prerequisites are met before debugging starts. Check the terminal for detailed information if there are issues with your environment. After this process, the Office desktop application launches and sideloads the add-in. Please note that the first time you run a project, it may make take a few minutes to install the dependencies. You'll need to install the certificate when prompted.
```

Rick-Kirkham commented on docs/includes/dev-kit-stop-debugging.md:
```diff
@@ -0,0 +1,7 @@
+
+1. Open the extension by selecting the Office Add-ins Development Kit icon in the **Activity Bar**.
+1. Select **Stop Previewing Your Office Add-in**. This closes the web server and removes the add-in from the registry and cache.
+1. Close the Office application window at your convenience.
```
They should not leave it open because any buttons connected with the add-in will still be visible, but they will no longer work.

```suggestion
1. Close the Office application window.
```

Rick-Kirkham commented on docs/includes/dev-kit-stop-debugging.md:
```diff
@@ -0,0 +1,7 @@
+## Stop testing your Office Add-in
+
+Once you are finished testing and debugging the add-in, close the add-in by following these steps.
```
This is important enough that we should emphasize it. There can be buggy behavior the next time they create an add-in if they don't shut down properly.

```suggestion
Once you are finished testing and debugging the add-in, *always* close the add-in by following these steps. (Closing the Office application or web server window doesn't reliably deregister the add-in.)
```

Rick-Kirkham commented on docs/quickstarts/excel-quickstart-jquery.md:
```diff
@@ -8,8 +8,38 @@ ms.localizationpriority: high
 # Build an Excel task pane add-in
 
-In this article, you'll walk through the process of building an Excel task pane add-in with Yo Office.
+In this article, you'll walk through the process of building an Excel task pane add-in. You'll use either the Office Add-ins Development Kit or the Yeoman generator to create your Office Add-in. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally. If you'd like to use Visual Studio Code, we recommend the Office Add-ins Developmen ...
```
```suggestion
In this article, you'll walk through the process of building an Excel task pane add-in. You'll use either the Office Add-ins Development Kit or the Yeoman generator to create your Office Add-in. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally. If you'd like to create the add-in project within Visual Studio Code, we recommend the Office Add-ins Development Kit.
```

Rick-Kirkham commented on docs/quickstarts/word-quickstart-yo.md:
```diff
@@ -8,8 +8,38 @@ ms.localizationpriority: high
 # Build your first Word task pane add-in
 
-In this article, you'll walk through the process of building a Word task pane add-in.
+In this article, you'll walk through the process of building a Word task pane add-in. You'll use either the Office Add-ins Development Kit or the Yeoman generator to create your Office Add-in. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally. If you'd like to use Visual Studio Code, we recommend the Office Add-ins Development  ...
```
```suggestion
In this article, you'll walk through the process of building a Word task pane add-in. You'll use either the Office Add-ins Development Kit or the Yeoman generator to create your Office Add-in. Select the tab for the one you'd like to use and then follow the instructions to create your add-in and test it locally. If you'd like to create the add-in project within Visual Studio Code, we recommend the Office Add-ins Development Kit.
```

Rick-Kirkham reviewed (changes requested):
Just a few small changes.

MingjiaLiu1995 commented:
> Just a few small changes.

@Rick-Kirkham Thanks! I have addressed all comments. Let's move the timeline topic to the email thread. It mainly depends on the sample migration repo status.

MingjiaLiu1995 commented:
Hi @Rick-Kirkham, how can I get the MS Learn preview pages of these changes? I would like to check the updated content work fine before it goes live. Thanks!

Rick-Kirkham commented:
> Hi @Rick-Kirkham, how can I get the MS Learn preview pages of these changes? I would like to check the updated content work fine before it goes live. Thanks!

Because the source version is in a different fork, the MS Learn preview doesn't work. The best you can do is try viewing it in GitHub (see screen shot below), but this will have different formatting. Another option is to clone (not fork) the repo from OfficeDev. Then create a new branch of the repo locally. Then copy your changes from the forked clone of the repo to the new branched clone. Then create a new PR from the branched clone and close the original PR. 

<img width="1172" alt="image" src="https://github.com/user-attachments/assets/cf234af5-05b9-46d7-8138-aadb5bd59933">
